### PR TITLE
Fix FlyteFS

### PR DESCRIPTION
**Pull request by pingsutw**
## Tracking issue
NA

## Why are the changes needed?
Failed to upload the FlyteFile to ABFS

## What changes were proposed in this pull request?
Add additional metadata to the request header

## How was this patch tested?
```
pyflyte --verbose run --remote test2.py wf --f test/test.csv --d test
```

```python
import pandas as pd

from flytekit import task, workflow
from flytekit.types.directory import FlyteDirectory
from flytekit.types.file import FlyteFile


@task()
def print_data_directory(f: FlyteFile, d: FlyteDirectory):
    df = pd.read_csv(f)
    print(d)
    print(df.head())


@workflow
def wf(f: FlyteFile, d: FlyteDirectory):
    print_data_directory(f=f, d=d)
```

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs
Blocked by https://github.com/flyteorg/flyte/pull/4971

## Docs link
NA

### CODE REVIEW

**Comment by eapolinario** [comment]
This solution looks like it's papering over the problem. Why is this specific to Azure? I have a feeling that either stow is not returning the etag in https://github.com/flyteorg/flyte/blob/6f5c47154549dd9f697f4a44547eda5197d40a52/flyteadmin/dataproxy/service.go#L83 or Azure is not returning the etag.

**Review by wild-endeavor — APPROVED** [review]

File: flytekit/remote/remote_fs.py [hunk 1]
```diff
@@ -278,6 +231,7 @@ async def _put(
         res = await super()._put(lpath, REMOTE_PLACEHOLDER, recursive, callback, batch_size, **kwargs)
         if isinstance(res, list):
             res = self.extract_common(res)
+        FlytePathResolver.add_mapping(rpath.strip("/"), res)
```
Comment (wild-endeavor) [inline flytekit/remote/remote_fs.py:234]: when is this necessary?   will this break on windows?

**Review by wild-endeavor — APPROVED** [review]
thanks!

File: flytekit/remote/remote_fs.py (hunk 1 above)
Comment (eapolinario) [reply flytekit/remote/remote_fs.py:234]: Let's use `os.path.sep` instead?

Comment (pingsutw) [reply flytekit/remote/remote_fs.py:234]: done

**Review by eapolinario — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**